OHM-Trade-Agent-v1/tools/opip_platform_restore_verify.py

```
@@ -0,0 +1,105 @@
+
+
+def safe_extract(archive: Path, destination: Path) -> None:
+    root = destination.resolve()
+    with tarfile.open(archive, "r:gz") as tar:
+        for member in tar.getmembers():
+            target = (destination / member.name).resolve()
+            if target != root and root not in target.parents:
+                raise RuntimeError(f"unsafe archive member: {member.name}")
+            if member.issym() or member.islnk():
+                raise RuntimeError(f"links are not permitted in backup archive: {member.name}")
+        tar.extractall(destination)
```

Ruff (0.16.2)

[error] Uses of `tarfile.extractall()`

(S202)